### ISSUE TITLE
Coverage build failure for lldb-eval

### ISSUE THREAD

**Issue by werat** [google/oss-fuzz#5909]
Hi, the coverage build for [lldb-eval](https://github.com/google/oss-fuzz/blob/master/projects/lldb-eval) is failing ([bug](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34923
)) and I can't figure out what the problem is.

Latest build -- https://oss-fuzz-build-logs.storage.googleapis.com/log-e32d5b77-7ff5-476e-a287-229147c251f8.txt

The error happens close to the end during `Step #5`.
```
Starting Step #5
Step #5: Already have image (with digest): gcr.io/oss-fuzz-base/base-runner
Step #5: Running lldb_eval_libfuzzer_test
Step #5: warning: /workspace/out/coverage/dumps/lldb_eval_libfuzzer_test.349514951254867133_0.profraw: Invalid instrumentation profile data (file header is corrupt)
Step #5: error: No profiles could be merged.
Step #5: [2021-06-10 11:41:39,546 INFO] Finding shared libraries for targets (if any).
Step #5: [2021-06-10 11:41:44,549 INFO] Finished finding shared libraries for targets.
Step #5: error: lldb_eval_libfuzzer_test, /workspace/out/coverage/lib/liblldb.so.11: Failed to load coverage: No such file or directory
Step #5: error: Could not load coverage information
Step #5: error: /workspace/out/coverage/dumps/*.profdata: No such file or directory
Step #5: ********************************************************************************
Step #5: Code coverage report generation failed.
Step #5: To reproduce, run:
Step #5: python infra/helper.py build_image lldb-eval
Step #5: python infra/helper.py build_fuzzers --sanitizer coverage lldb-eval
Step #5: python infra/helper.py coverage lldb-eval
Step #5: ********************************************************************************
Finished Step #5
ERROR
ERROR: build step 5 "gcr.io/oss-fuzz-base/base-runner" failed: step exited with non-zero status: 1
```

The first error looks like some tooling problem maybe? I see similar reports in #3065
```
Step #5: warning: /workspace/out/coverage/dumps/lldb_eval_libfuzzer_test.349514951254867133_0.profraw: Invalid instrumentation profile data (file header is corrupt)
Step #5: error: No profiles could be merged.
```

The second error complains about `liblldb.so.11`. The fuzzer target works fine, so the library must be there. One potential issue is that it's a symlink `liblldb.so.11 -> liblldb.so.11.1.0`, where `liblldb.so.11.1.0` is the actual lib.

```
Step #5: error: lldb_eval_libfuzzer_test, /workspace/out/coverage/lib/liblldb.so.11: Failed to load coverage: No such file or directory
Step #5: error: Could not load coverage information
Step #5: error: /workspace/out/coverage/dumps/*.profdata: No such file or directory
```

I've tried reproducing the build locally using the commands above and it works fine:
```
> git log -n 1
commit 73d78b88790b501f119801c4f68463180b76e1d9 (HEAD -> master, origin/master, origin/HEAD)
...

> python infra/helper.py build_image lldb-eval
Pulling latest base images...
Running: docker build --no-cache -t gcr.io/oss-fuzz/lldb-eval projects/lldb-eval
...

> python infra/helper.py build_fuzzers --sanitizer coverage lldb-eval
Running: docker build -t gcr.io/oss-fuzz/lldb-eval projects/lldb-eval
...

> python infra/helper.py coverage lldb-eval
Downloading corpora for lldb-eval project to $HOME/src/fuzz/oss-fuzz/build/corpus/lldb-eval
Running: docker run --rm --privileged -i -e FUZZING_ENGINE=libfuzzer -e FUZZING_LANGUAGE=c++ -e PROJECT=lldb-eval -e SANITIZER=coverage -e HTTP_PORT=8008 -e COVERAGE_EXTRA_ARGS= -p 8008:8008 -v $HOME/src/fuzz/oss-fuzz/build/corpus/lldb-eval:/corpus -v $HOME/src/fuzz/oss-fuzz/build/out/lldb-eval:/out -t gcr.io/oss-fuzz-base/base-runner coverage
Running lldb_eval_libfuzzer_test
...
Serving the report on http://127.0.0.1:8008/linux/index.html
Serving HTTP on 0.0.0.0 port 8008 ...
^C
Keyboard interrupt received, exiting.
Successfully generated clang code coverage report.
```
I can upload full local build log if needed.

Can you help us figure this out? Thanks!

**Comment by jonathanmetzman** [comment]
I can get it working myself.
Can you try again

**Comment by werat** [comment]
In my local build I pulled the latest images in the beginning. Right now `python3 infra/helper.py pull_images` says that all images are up to date. I've tried again, the build still works fine locally for me.

What do you mean by `I can get it working myself without these issues`? Does it mean the coverage build works fine locally for you too?

**Comment by jonathanmetzman** [comment]
Sorry, my answer was way too short.
I actually can repro your issue.
But I was able to get around it by creating a corpus:
```
mkdir -p build/corpus/lldb-eval/lldb_eval_libfuzzer_test/
echo "" > build/corpus/lldb-eval/lldb_eval_libfuzzer_test/1
python infra/helper.py coverage --no-corpus lldb-eval
```

I think the corpus step fails (for you and I, not on the infra) because lldb-eval is new enough that there is no public (ie: older than 90 days) backup. But I'm not sure why this leads to that shared library error which I also get without a corpus.

**Comment by werat** [comment]
So when you ran the coverage build locally it also failed for you? I've tried removing all the local data and running the coverage build again with `--no-corpus` -- the build is still successful:
```
> python infra/helper.py build_image lldb-eval
> python infra/helper.py build_fuzzers --sanitizer coverage lldb-eval
> mkdir -p build/corpus/lldb-eval/lldb_eval_libfuzzer_test/
> echo "" > build/corpus/lldb-eval/lldb_eval_libfuzzer_test/1
> python infra/helper.py coverage --no-corpus lldb-eval
...
Successfully generated clang code coverage report.
```

Is local build with `python infra/helper.py coverage` somehow different from the infra one?

**Comment by jonathanmetzman** [comment]
>So when you ran the coverage build locally it also failed for you?

Yes.


>I've tried removing all the local data and running the coverage build again with --no-corpus -- the build is still successful:

Right I found this too.


>Is local build with python infra/helper.py coverage somehow different from the infra one?

It's different in multiple ways. One way it is different (which is why I think my above explanation doesn't apply) is it has permission to get the latest corpus from OSS-Fuzz, so it won't have the same corpus issue we did (which is very weird IMO and I don't understand why it happens with your fuzzer).

The other way it is different is it doesn't actually use the same commands we did. I think it passes the docker commands for Google cloud build  in this script (https://github.com/google/oss-fuzz/blob/master/infra/build/functions/build_and_run_coverage.py#L140)
I can try to see why this fails for your project, but I might not get the time for a while.
